Library/Homebrew/extend/os/linux/install.rb

```diff
@@ -76,23 +76,43 @@ def symlink_ld_so
     end
     private_class_method :symlink_ld_so
 
-    def symlink_gcc_libs
+    def setup_preferred_gcc_libs
       gcc_opt_prefix = HOMEBREW_PREFIX/"opt/#{OS::LINUX_PREFERRED_GCC_FORMULA}"
+      ld_so = HOMEBREW_PREFIX/"opt/glibc/lib/ld-linux-x86-64.so.2"
+
+      if ld_so.readable?
+        ld_so_conf_d = HOMEBREW_PREFIX/"etc/ld.so.conf.d"
+        ld_so_conf_d.mkpath
+        ld_so_conf_d.chmod "go-w"
+
+        # Add gcc to ld search paths
+        ld_gcc_conf = ld_so_conf_d/"50-homebrew-preferred-gcc.conf"
+        ld_gcc_conf.atomic_write <<~EOS
+          #{gcc_opt_prefix}/lib/gcc/#{GCC_VERSION_SUFFIX}
+        EOS
+        ld_gcc_conf.chmod "u=rw,go-wx"
```

Comment:
Should use `FileUtils.chmod mode, path` instead:

```suggestion
        chmod "u=rw,go-wx", ld_gcc_conf
```